index.js

```diff
@@ -1981,6 +1983,56 @@
     return result;
   }
 
+  //# takeWhile :: (Applicative f, Foldable f, Monoid (f a)) => (a -> Boolean, f a) -> f a
+  //.
+  //. Discards the first inner value which does not satisfy the predicate, and
+  //. all subsequent inner values.
+  //.
+  //. This function is derived from [`concat`](#concat), [`empty`](#empty),
+  //. [`of`](#of), and [`reduce`](#reduce).
```

Comment:
Is there a reason you're referring to the lowest level functions, rather than just mentioning "filter"?

Reply:
Maybe to explain the class constraints?

Reply:
> Is there a reason you're referring to the lowest level functions, rather than just mentioning "filter"?

I see three reasons:

  - It is consistent with our descriptions of other derived functions. This is not a compelling reason, of course, as we could easily update these.

  - It assists readers in determining that derived functions are derived from Fantasy Land methods exclusively. Readers familiar with Fantasy Land need not follow internal links.

  - The decision to derive a function from underlying Fantasy Land methods *directly* or *indirectly* should be an implementation detail. Switching from one approach to the other should not necessitate documentation changes.